skfda/preprocessing/registration/validation.py

```diff
@@ -0,0 +1,479 @@
+
+    .. math::
+
+        C_R = 1 + \frac{\frac{1}{N}\sum_{i}^{N}\int [Dh_i(t)-\overline{Dh}(t)]
+        [ y_i^2(t)- \overline{y^2}(t) ]dt}
+        {\frac{1}{N} \sum_{i}^{N} \int y_i^2(t)dt}
+
+    whose structure is related to the covariation between the deformation
+    functions :math:`Dh_i(t)` and the squared registered functions
+    :math:`y_i^2(t)`. When these two sets of functions are independents
+    :math:`C_R=1`, as in the case of shift registration.
+
+    The total mean square error is decomposed in the two sources of
+    variability.
+
+    .. math::
+        \text{MSE}_{total} = \text{MSE}_{amp} + \text{MSE}_{phase}
+
+    The squared multiple correlation index of the proportion of the total
+    variation due to phase is defined as:
+
+    .. math::
+        R^2 = \frac{\text{MSE}_{phase}}{\text{MSE}_{total}}
+
+    See [KR08-3]_ for a detailed explanation.
+
+
+    Args:
+        X (:class:`FData`): Unregistered functions.
+        y (:class:`FData`, optional): Target data, generally the same as X. By
+            default 'None', which uses `X` as target.
+        return_stats (boolean, optional): If `true` returns a named tuple
+            with four values: :math:`R^2`, :math:`MSE_{amp}`, :math:`MSE_{pha}`
+            and :math:`C_R`. Otherwise the squared correlation index
+            :math:`R^2` is returned. Default `False`.
+        eval_points: (array_like, optional): Set of points where the
+            functions are evaluated to obtain a discrete representation and
+            perform the calculation.
+
+    Returns:
+        (float or :class:`NamedTuple <typing.NamedTuple>`): squared correlation
+        index :math:`R^2` if `return_stats` is `False`. Otherwise a named
+        tuple containing:
+
+            * `r_squared`: Squared correlation index :math:`R^2`.
+            * `mse_amp`: Mean square error of amplitude
+              :math:`\text{MSE}_{amp}`.
+            * `mse_pha`: Mean square error of phase :math:`\text{MSE}_{pha}`.
+            * `c_r`: Constant :math:`C_R`.
+
+
+    Raises:
+        ValueError: If the functional data is not univariate.
+
+    References:
+        ..  [KR08-3] Kneip, Alois & Ramsay, James. (2008).  Quantifying
+            amplitude and phase variation. In *Combining Registration and
+            Fitting for Functional Models* (pp. 14-15). Journal of the American
+            Statistical Association.
+        ..  [RGS09-8-5] Ramsay J.O., Giles Hooker & Spencer Graves (2009). In
+            *Functional Data Analysis with R and Matlab* (pp. 125-126).
+            Springer.
+
+    See also:
+        :class:`RegistrationScorer <RegistrationScorer>`
+        :func:`least_squares <least_squares>`
+        :func:`sobolev_least_squares <sobolev_least_squares>`
+        :func:`pairwise_correlation <pairwise_correlation>`
+
+    """
+    from scipy.integrate import simps
+
+    # Parameter checks
+    if not X._univariate or not y._univariate:
```

Comment:
Where is `_univariate` defined?

Reply:
Ups, It is an error during the separation of the PR.

I was going to comment on it in the next PR, since the check that an object is univariate is repeated a lot, I had thought of adding a property or method in FData to check it and avoid code repetition, but I don't know if it is more useful a property "_univariate" or a method "_check_univariate()".

Reply:
I think it is better a function, not a property nor a method.

Reply:
Added in e2ff4a6.